libs/dmlf/include/dmlf/distributed_learning/utilities/mnist_client_utilities.hpp

```diff
@@ -18,11 +18,12 @@
 //------------------------------------------------------------------------------
 
 #include "dmlf/distributed_learning/distributed_learning_client.hpp"
-#include "ml/dataloaders/mnist_loaders/mnist_loader.hpp"
+#include "ml/dataloaders/tensor_dataloader.hpp"
```

Comment:
I think with a little careful thought you could perhaps delete this file altogether?

The main difference between this and the boston_client_utilities is that Boston takes in a Tensor and MNist takes in a filename. If you invoke the mnist file reading utilities you've created in the example main.cpp files perhaps you can just invoke the Boston client utilities and be done with it? (you may need to make some other small changes, such as specifying the model architecture somewhere)